homeassistant/components/met/weather.py

```diff
@@ -109,43 +101,35 @@ class MetWeather(SingleCoordinatorWeatherEntity[MetDataUpdateCoordinator]):
 
     def __init__(
         self,
+        hass: HomeAssistant,
```

Comment:
I think this was the way it was because we aren't supposed to pass `hass` for entity construction and let it get set by `async_added_to_hass`

Reply:
Now we have a reason